cmake/Modules/thread-local.cmake

```diff
@@ -0,0 +1,25 @@
+include(CheckCXXSourceCompiles)
+
+CHECK_CXX_SOURCE_COMPILES("
+static thread_local int tls;
+int main(void)
+{
+  return 0;
+}" HAVE_THREAD_LOCAL)
```

Comment:
Because we live in the real world.
Even the latest(as used by travis-ci osx builds) XCode compiler does not support it.